virtcontainers/qemu_arch_base.go

```diff
@@ -110,6 +111,8 @@ const (
 	maxDevIDSize            = 31
 )
 
+const bridgePCIStartAddr = 5
```

Comment:
Why 5 ? I thought Qemu was reserving only the slot 0 and 1. Could you add a comment explaining this ?

Reply:
@sboeuf  Slot 0 and 1 are reserved in case of bridges. I am choosing address 5 here on the main pci bus. There are a few devices that we add on the main bus currently, eg
```
sudo docker run -it --runtime=kata-runtime  centos/tools
[root@e9f831b1e6e7 /]# lspci
00:00.0 Host bridge: Intel Corporation 440FX - 82441FX PMC [Natoma] (rev 02)
00:01.0 ISA bridge: Intel Corporation 82371SB PIIX3 ISA [Natoma/Triton II]
00:01.1 IDE interface: Intel Corporation 82371SB PIIX3 IDE [Natoma/Triton II]
00:01.3 Bridge: Intel Corporation 82371AB/EB/MB PIIX4 ACPI (rev 03)
00:02.0 Communication controller: Red Hat, Inc Virtio console
00:03.0 PCI bridge: Red Hat, Inc. QEMU PCI-PCI bridge
00:04.0 SCSI storage controller: Red Hat, Inc Virtio SCSI
00:05.0 Unclassified device [0002]: Red Hat, Inc Virtio filesystem
00:06.0 Ethernet controller: Red Hat, Inc Virtio network device
```
Here pci bridge is added at 00:03.0, I just chose to add bridges starting at 00.04.0, with the other devices occupying the other slots.

Reply:
@sboeuf We dont care about the location of other devices, we just care about the bridge address. Bridges are cold-plugged, the first available slot in the current sequence is 0x03,  I could have just chosen that. By simply choosing 0x5,  I am adding bridges from that address, leaving the other addresses for other devices, qemu will find the first available slot for those devices.

Reply:
I totally get it, I was wondering if other cold plugged devices (do we have some ?), could be cold plugged before the bridge. If that's the case, then you cannot guarantee that nothing else will appear at address 5.
Following this explanation, if you're sure that the bridge is gonna be the first thing being added to the main bridge, then I think it is better to take the first PCI address that we know available (and explain that previous addresses 0 and 1 are indeed used internally by Qemu, but this is a known thing).

Reply:
I added a buffer to have some flexibility. i can change this from 0x5 to 0x3 if it makes things any clearer.

Reply:
@sboeuf Changes pushed following our conversation. I have added in comments and moved the bridge append early on. ptal.